pkg/controller/validate/validate.go

```diff
@@ -155,32 +160,42 @@ func moreThanOneIsNonNil(values ...interface{}) bool {
 }
 
 func instanceSizeForAdvancedDeployment(replicationSpecs []*mdbv1.AdvancedReplicationSpec) error {
-	var instanceSize string
-	err := errors.New("instance size must be the same for all nodes in all regions and across all replication specs for advanced deployment ")
+	err := errors.New("instance size must be the same for all nodes in all regions and across all replication specs for advanced deployment")
+
+	instanceSize := ""
+	firstNonEmptySize := func(region *mdbv1.AdvancedRegionConfig) string {
+		if instanceSize != "" {
+			return instanceSize
+		}
 
-	isInstanceSizeEqual := func(nodeInstanceType string) bool {
-		if instanceSize == "" {
-			instanceSize = nodeInstanceType
+		if region.ElectableSpecs != nil && region.ElectableSpecs.InstanceSize != "" {
+			return region.ElectableSpecs.InstanceSize
 		}
 
-		return nodeInstanceType == instanceSize
+		if region.ReadOnlySpecs != nil && region.ReadOnlySpecs.InstanceSize != "" {
+			return region.ReadOnlySpecs.InstanceSize
+		}
+
+		if region.AnalyticsSpecs != nil && region.AnalyticsSpecs.InstanceSize != "" {
+			return region.AnalyticsSpecs.InstanceSize
+		}
+
+		return ""
 	}
 
 	for _, replicationSpec := range replicationSpecs {
 		for _, regionSpec := range replicationSpec.RegionConfigs {
-			if instanceSize == "" {
-				instanceSize = regionSpec.ElectableSpecs.InstanceSize
-			}
+			instanceSize = firstNonEmptySize(regionSpec)
```

Comment:
Is it legal for any region, including the first, not to specify a instance size at all anywhere?

Reply:
If `nodeCount`is set to zero, instanceSize doesn't actually matter, otherwise, it's not allowed. I'll make sure I test this case.